src/functions/authoriser/framework/handler.ts

```diff
@@ -1,6 +1,49 @@
-import { CustomAuthorizerEvent, CustomAuthorizerResult, Context } from 'aws-lambda';
+import * as util from 'util';
+import { CustomAuthorizerEvent, CustomAuthorizerResult } from 'aws-lambda';
+import * as jwksClient from 'jwks-rsa';
+import nodeFetch from 'node-fetch';
+import { AdJwtVerifier } from '../application/AdJwtVerifier';
 
-const newCustomAuthorizerResult = (principalId: string, effect: string, resource: string)
+const ensureNotNullOrEmpty = (val: any, fieldName: string) => {
+  if (val === null || val === undefined || val === '') {
+    throw `Error: ${fieldName} is null or empty`;
+  }
+};
+
+/**
+ * Creates an AdJwtVerifier that can verify the authenticity of Azure Active Directory JWTs.
+ */
+const createAdJwtVerifier = async (): Promise<AdJwtVerifier> => {
+  const tenantId: string = process.env.DVSA_MES_AzureAD_TenantId || '';
+  const applicationId: string = process.env.DVSA_MES_AzureAD_ClientId || '';
+
+  ensureNotNullOrEmpty(tenantId, 'tenantId');
+  ensureNotNullOrEmpty(applicationId, 'applicationId');
+
+  const openidConfig = await (await nodeFetch(
```

Comment:
do we need to await twice? is `.json()` not just a synchronous operation on the response?

Reply:
The `json` method returns a promise too!  Each promise needs awaiting before it can be changed.  If there's a better way, please do let me know!  I didn't like to double `await` either :)